src/Extension/ExtendableTrait.php

```diff
@@ -412,12 +403,11 @@ public function extendableSet($name, $value)
         $parent = $this->extensionGetParentClass();
         if ($parent !== false && $this->extensionMethodExists($parent, '__set')) {
             $this->extensionCallMethod($parent, '__set', [$name, $value]);
+            return;
         }
 
-        /*
-         * Setting an undefined property
-         */
-        if (!self::$extendableGuardProperties) {
+        // local dynamic property
+        if (array_key_exists($name, $this->getDynamicProperties())) {
```

Comment:
@mjauvin what's this condition needed for?

Reply:
It prevents creating a dynamic property just by setting it (if there's no parent `__set` method for example)